client/icecc-create-env.in

```diff
@@ -336,9 +337,24 @@ if test -f /etc/ld.so.conf; then
     fi
     echo "$directive $path"
   done </etc/ld.so.conf >$tmp_ld_so_conf
+
+  # Work around new directory layout for clang in ubuntu >= 16.10
+  if [ -n "$clangprefix" ]; then
+      echo "$clangprefix/lib" >> $tmp_ld_so_conf
+  fi
```

Comment:
Why would this be necessary? Doesn't clang have this directory in ld.so.conf? And if not, how does it work normally?

Reply:
It was a combination of clang being a symlink into a directory that is a symlink, using RPATH=../lib (or something similar), the regex-based tricks to resolve /../ in paths, and icecream requiring that clang be at /bin/clang rather than /the/real/path/to/clang. With all of that, the simplest and most robust solution seems to just be listing every lib dir in the ld.so.conf.